roles/openshift_sanitize_inventory/tasks/main.yml

```diff
@@ -77,6 +77,23 @@
         openshift_cloudprovider_vsphere_host is undefined or openshift_cloudprovider_vsphere_datacenter is undefined or
         openshift_cloudprovider_vsphere_datastore is undefined )
 
+- name: Ensure azure configuration variables are defined
+  block:
+    - name: ensure provider configuration variables are defined
+      fail:
+        msg: "'Required variable for azure provider '{{ item }}' is not defined"
+      when: vars[item] is not defined
+      with_items:
```

Comment:
No need for with_items here.  Better to check that all are defined at once and provide link to docs with required vars.

```yaml
when: >
  openshift_cloudprovider_azure_client_id is not defined
  or ... is not defined
  or ..
```

Reply:
Ah. Can change that to recommendation. Just used with items as it'll provide a better error message in this case either all items that failed.